packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -21,12 +20,32 @@ import {
 } from "@fluidframework/test-utils";
 import { describeNoCompat, ITestDataObject, itExpects, TestDataObjectType } from "@fluidframework/test-version-utils";
 import { delay, stringToBuffer } from "@fluidframework/common-utils";
-import { IContainer, IErrorBase, LoaderHeader } from "@fluidframework/container-definitions";
-import { IRequest } from "@fluidframework/core-interfaces";
+import { IContainer, LoaderHeader } from "@fluidframework/container-definitions";
+import { IFluidHandle, IRequest } from "@fluidframework/core-interfaces";
 import { ISummaryTree } from "@fluidframework/protocol-definitions";
 import { validateAssertionError } from "@fluidframework/test-runtime-utils";
+import { IFluidDataStoreChannel } from "@fluidframework/runtime-definitions";
 import { getGCStateFromSummary, getGCTombstoneStateFromSummary } from "./gcTestSummaryUtils";
 
+/**
+ * Potential GC Recovery handle
+ */
+class RecoveryHandle implements IFluidHandle {
```

Comment:
I am not sure I understand why this is added and what its used for.

Reply:
The recovery handle is just for recovering a tombstoned datastore. If a container is in a bad state, all we want a customer to do is store the handle. We do not want them to do any more with that specific container, throwing on `get` is one way of doing so. Once the summarizer has summarized, we can load from the fixed summary and then the datastore will be properly referenced and we would return a RemoteObjectHandle, which has the proper things to do. Otherwise, if the customer stores the handle incorrectly, they should find out by retreiving the datastore and continuing to get tombstoned errors.